ambassador/ambassador/ir/irmappingfactory.py

```diff
@@ -36,5 +36,12 @@ def finalize(cls, ir: 'IR', aconf: Config) -> None:
         # OK. We've created whatever IRMappings we need. Time to create the clusters
         # they need.
 
+        to_delete = []
         for group in ir.groups.values():
-            group.finalize(ir, aconf)
```

Comment:
this code is just the most straightforward solution. feedback welcome on the way it should be done.

Reply:
Pretty sure I've done similar things elsewhere, actually. Looks sane.